### PR TITLE
ref(app-platform): Add issue open in link schema option

### PR DESCRIPTION
**Pull request by MeredithAnya**
This adds another UI augmentation schema with `type: stacktrace-link`. The idea would be that one could click a button (or from a dropdown) where they'd want to open the event data

**attributes**

| attribute | required  | description  |
|---|---|---|
| `uri`  | yes  | where we post the data to in the other service where you are 'opening' the issue in.  |
| `params`  | no  |  the kinds of data that will be in the request*  |

*By default the `installation_id` and `project` (slug) is sent in the params, and for now the only options we'll send in addition to that will be `filename`, and `lineno`.

**example**
```javascript
    {
	type: "stacktrace-link".
	uri: "/something",
	params: ["filename", "lineno"] 
    }
```

### CODE REVIEW

**Review by mnoble — COMMENTED** [review]
I don't think we need the `-link` part in this element. That seems very specific to how the feature is implemented – when I think the names of these components should be more based around the feature they're enabling. Maybe just `issue-open-in`?

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 1]
```diff
@@ -242,6 +242,27 @@
...
+                    'enum': ['issue-open-in-link'],
+                },
+                'uri': {
+                    '$ref': '#/definitions/uri',
+                },
+                'params': {
+                    'type': 'array',
+                    'items': {
+                        'type': 'string',
+                        'enum': ['project', 'filename', 'lineno']
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/api/validators/sentry_apps/schema.py]: eh maybes, do you think it makes sense to include org slug too then?

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 2]
```diff
@@ -242,6 +242,27 @@
...
+                'type': {
+                    'type': 'string',
+                    'enum': ['issue-open-in-link'],
+                },
+                'uri': {
+                    '$ref': '#/definitions/uri',
+                },
+                'params': {
+                    'type': 'array',
+                    'items': {
```
[the remark this answers is not in the document]
Comment (MeredithAnya) [reply src/sentry/api/validators/sentry_apps/schema.py]: yeah. although I don't know why someone would just put an empty array so maybe it's worth adding the `minItems: 1`?

[the remark this answers is not in the document]
Comment (mnoble) [reply src/sentry/api/validators/sentry_apps/schema.py]: If they do, it's fine. Shouldn't harm anything.

File: src/sentry/api/validators/sentry_apps/schema.py [hunk 3]
```diff
@@ -244,6 +244,28 @@
...
+            'type': 'object',
+            'properties': {
+                'type': {
+                    'type': 'string',
+                    'enum': ['stacktrace-link'],
+                },
+                'uri': {
+                    '$ref': '#/definitions/uri',
+                },
+                'params': {
```
Comment (mnoble) [inline src/sentry/api/validators/sentry_apps/schema.py]: If we're scoping to a line in a stacktrace, I don't know that we need to make `params` configurable any longer. I don't know what other than a filename and line number you can derive from a line of code.

**Review by mnoble — APPROVED** [review]
👍

**Pull request merged**